errors/env-key-not-allowed.md

```diff
@@ -0,0 +1,9 @@
+# The key "<your key>" under "env" in next.config.js is not allowed.
+
+#### Why This Error Occurred
+
+Next.js configures internal variables for replacement itself. These start with `__` or `NODE_`, for this reason they are not allowed as values for `env` in `next.config.js`
```

Comment:
This might be better as:

> Next.js uses special environment variables to configure itself. Because of this, you are not allowed to use values that start with `__` or `NODE_` for `env` in `next.config.js`.